qa/suites/orch/cephadm/smoke-roleless/2-services/iscsi.yaml

```diff
@@ -3,6 +3,27 @@ tasks:
     host.a:
       - ceph osd pool create foo
       - rbd pool init foo
-      - ceph orch apply iscsi foo u p
+      - ceph orch apply iscsi foo u p --placement "*"
```

Comment:
Can you explain what this does and why this is needed for this test?

Reply:
It puts an iscsi on every host instead of the default which is just on any 2 random hosts in the cluster. This is just to make it so there's always an iscsi on whatever "host.a" is to make the testing a bit easier. Otherwise we'd have to check which hosts got an iscsi and make sure to run the checks there.

Reply:
Wouldn't it make more sense to place on `host.a` (or `host.a` and `host.b`) explicitly to ensure that it's self-evident?

Reply:
I don't know what `host.a` is going to be. Cephadm, during these tests, will know the hosts as things like `smithi01` or `smithi72` so doing explicit placement would require providing that hostname. There might be some way through teuthology to figure out what actual host `host.a` is at runtime, but I don't know how to do it. If you know how to do that (or you know somebody who knows I can ask) then I'd agree that would make more sense. If not, given this is just a smoke test that makes sure the iscsi daemons deploy without going into error state plus this new check for the /etc/hosts, I don't think placing one on every host invalidates the test in any way.

Reply:
will add that this is only because this test is explicitly not using roles (hence `smoke-roleless` directory name) where you can specify the daemons for `host.a` in the yaml. It's meant to include testing of the apply commands which don't know anything about the `host.a`, `host.b` abstraction teuthology provides

Reply:
I see.  Honestly, it's a bit confusing and I'm also not a fan of embedding actual test logic in the yaml definition this way.

I'd place the shell snippet into its own file, `qa/workunits/cephadm/test_iscsi_etc_hosts.sh` for example.  You said you didn't want to add a whole new test for this so I propose wiring it up from the existing `qa/suites/orch/cephadm/workunits/task/test_iscsi_pids_limit/test_iscsi_pids_limit.yaml`:
```
      client.0:
        - cephadm/test_iscsi_pids_limit.sh
        - cephadm/test_iscsi_etc_hosts.sh
```
This file could then be renamed to `test_iscsi_container.yaml`.

Reply:
I like that. Moved things over and did the renaming